### PR TITLE
GA bug fixes

### LINKED ISSUE #330: BUG - Submitting empty feedback is buggy
Sending empty feedback form glitches the response message, it shows for just a fraction of a second, and then it disappears:

glitch send

What I would suggest is that if the message field is empty, it just closes the dialog (so like close button does). Or we disable the Send button until there is some message, that could work too.

### PR DESCRIPTION
**Pull request by unseen1980**
# Bug fixes for GA 
 
## Issues resolved
#330, #299, #337, #331, #328  #327 

## Description

#330 submitting feedback is fixed
Screenshot 2022-12-12 at 16 41 08

#299 (Bug Search Page, November) -> All bugs are fixed except the sticky header 
#337 (Info Icon squished)
#331 (Feedback Icon)
#328 (Home Icon)
#327 (Is this content helpful icon)

## Checklist for Pull Requests

- [x] Supplied as many details as possible on this change
- [x] Included the link to the associated issue in the Issue section above
- [x] The code conforms to the general [development guidelines](https://github.com/SAP/guided-answers-extension/blob/main/docs/developer-guide.md).
- [ ] The code has **unit tests** where applicable and is easily unit-testable
- [x] This branch is appropriately named for the associated issue

### CODE REVIEW

**Comment by sanat-thukral** [comment]
This PR contains bug fixes for:
#299, #337, #331, #330, #329, #328  #327 

However, it doesn't cover the bug in #299 where the GA Header isn't sticky while scrolling

File: packages/webapp/src/webview/ui/components/App/App.scss [hunk 1]
```diff
@@ -37,6 +37,7 @@ h1 {
     display: flex;
     width: 100%;
     flex-direction: column;
+    // overflow: auto;
```
Comment (Klaus-Keller) [inline packages/webapp/src/webview/ui/components/App/App.scss:40]: Is this a left over?

File: packages/webapp/package.json [hunk 2]
```diff
@@ -51,7 +51,9 @@
         "@fluentui/react-focus": "8.8.8",
         "@fluentui/style-utilities": "8.8.1",
         "@fluentui/utilities": "8.13.2",
-        "@sap-ux/ui-components": "1.0.16",
-        "react-infinite-scroll-component": "6.1.0"
+        "@sap-ux/ui-components": "1.1.1",
+        "@types/react-transition-group": "4.4.5",
```
Comment (Klaus-Keller) [inline packages/webapp/package.json:55]: Why do we need `react-transition-group` and if we need it, can we have the types as `devDependency`?

File: packages/webapp/src/webview/ui/components/App/App.scss (hunk 1 above)
Comment (sanat-thukral) [reply packages/webapp/src/webview/ui/components/App/App.scss:40]: yup, I'll remove it. Thanks for point it out 😊

File: packages/webapp/package.json (hunk 2 above)
Comment (sanat-thukral) [reply packages/webapp/package.json:55]: Thanks, it's not required. I'll do a complete cleanup before submitting

**Comment by sanat-thukral** [comment]
@Klaus-Keller I wanted to check if I can update the version of `@sap-ux/ui-components` we are using in the repo in this PR or if should I do it another way. As I'm getting conflicts in the `pump-lock.yaml`

**Comment by Klaus-Keller** [comment]
> @Klaus-Keller I wanted to check if I can update the version of `@sap-ux/ui-components` we are using in the repo in this PR or if should I do it another way. As I'm getting conflicts in the `pump-lock.yaml`

@sanat-thukral 
Yes, you can update the used version of `@sap-ux/ui-components`.

**Comment by Klaus-Keller** [comment]
@sanat-thukral, could you please check the reported code smell: https://sonarcloud.io/project/issues?id=SAP_guided-answers-extension&pullRequest=335&resolved=false&types=CODE_SMELL ?

**Comment by sanat-thukral** [comment]
Thanks, @Klaus-Keller, that's fixed now 😄

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -101,8 +101,13 @@ importers:
       '@fluentui/react-focus': 8.8.10
       '@fluentui/style-utilities': 8.8.3
       '@fluentui/utilities': 8.13.4
+<<<<<<< HEAD
```
Comment (Klaus-Keller) [inline pnpm-lock.yaml:104]: Could it be that some merge resolution left-overs got into this file

File: packages/webapp/src/webview/ui/components/FeedbackSection/FeedbackSection.tsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ import { MessageDialogBox } from '../DialogBoxes/MessageDialogBox';
 import type { AppState } from '../../../types';
 import type { GuidedAnswerNodeId, GuidedAnswerTreeId } from '@sap/guided-answers-extension-types';
 import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';
-
+//test
```
Comment (Klaus-Keller) [inline packages/webapp/src/webview/ui/components/FeedbackSection/FeedbackSection.tsx:12]: What is this? Leftover?

File: pnpm-lock.yaml (hunk 3 above)
Comment (sanat-thukral) [reply pnpm-lock.yaml:104]: yeah I did have multiple merge conflicts

File: packages/webapp/src/webview/ui/components/FeedbackSection/FeedbackSection.tsx (hunk 4 above)
Comment (sanat-thukral) [reply packages/webapp/src/webview/ui/components/FeedbackSection/FeedbackSection.tsx:12]: Yes, that shouldn't be there 😅

File: packages/webapp/src/webview/ui/components/UIComponentsLib/Icons/Icons.tsx [hunk 5]
```diff
@@ -654,6 +656,22 @@ export const initIcons = function (): void {
                     />
                 </svg>
             ),
+            [UiIcons.MessageSent]: (
```
Comment (Klaus-Keller) [inline packages/webapp/src/webview/ui/components/UIComponentsLib/Icons/Icons.tsx:659]: Hi @sanat-thukral, in course of ticket https://github.com/SAP/guided-answers-extension/issues/364, @unseen1980 made sure that all icons we need are available in `@sap-ux/ui-components`. 

PR: https://github.com/SAP/guided-answers-extension/pull/387 removes now the local copy of icons component from this repo. I think we need to do the same for these icons here.

File: packages/webapp/test/DialogBoxes/FeedbackDialogBox.test.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        expect(actions.feedbackStatus).toHaveBeenCalled();
+    });
+
+    it('Should test send button', () => {
+        const { container } = render(
+            <Provider store={store}>
+                <FeedbackDialogBox />
+            </Provider>
+        );
+        // const setTextFieldHasValueMock = jest.fn();
```
Comment (Klaus-Keller) [inline packages/webapp/test/DialogBoxes/FeedbackDialogBox.test.tsx:55]: What about these commented lines?

File: packages/webapp/test/DialogBoxes/FeedbackSendDialogBox.test.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    const store = mockStore(initialState);
+
+    it('Should render a FeedbackSendDialogBox component', () => {
+        const { container } = render(
+            <Provider store={store}>
+                <FeedbackSentDialogBox />
+            </Provider>
+        );
+        expect(container).toMatchSnapshot();
+        // jest.runAllTimers();
```
Comment (Klaus-Keller) [inline packages/webapp/test/DialogBoxes/FeedbackSendDialogBox.test.tsx:39]: also commented lines of code

**Review by Klaus-Keller — COMMENTED** [review]
Hi @sanat-thukral,
Please see my comments.

File: packages/webapp/src/webview/ui/components/UIComponentsLib/Icons/Icons.tsx (hunk 5 above)
Comment (sanatcodes) [reply packages/webapp/src/webview/ui/components/UIComponentsLib/Icons/Icons.tsx:659]: Thanks for pointing that out Klaus, makes sense.

File: packages/webapp/test/DialogBoxes/FeedbackDialogBox.test.tsx (hunk 6 above)
Comment (sanatcodes) [reply packages/webapp/test/DialogBoxes/FeedbackDialogBox.test.tsx:55]: Thanks I'll remove those

**Review by Klaus-Keller — APPROVED** [review]
Thank you @sanat-thukral 
- code looks good and is easy to understand
- test coverage is good
- did locally build and test changes

Approved from my side.

**Pull request merged**